Yoast/javascript
# Fixes analysis for schema blocks.

## Summary


This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Fixes a bug where not all block issues were logged in the schema analysis in the block sidebar.
* [@yoast/schema-blocks] Improves the logger with colors and the ability to group log statements.
* [@yoast/schema-blocks] Adds a `required` property to the `VariationPicker` block instruction. If this is set to `true`, the block should have a variation selected to be considered valid.

## Relevant technical choices:

* I improved the logger to be able to get a better overview of the validation results.
	* Now with colors!
	* And the ability to group logging statements.
* I added named constructors for the `BlockValidationResult`:
  ```ts
  BlockValidationResult.Valid( blockInstance );
  ```
  is a bit more readable than
  ```ts
  new BlockValidationResult( blockInstance.clientId, blockInstance.name, BlockValidation.Valid );
  ```

## Test instructions

This PR can be tested by following these steps:

### Testing if the schema analysis gives a red result
* Companion PR of https://github.com/Yoast/wordpress-seo-premium/pull/3196, please follow its test instructions.

### Testing the improved logger
* Create a post.
* Add a job posting block.
* Check the browser console.
	* You should **not** see any debug statements for the schema blocks functionality.
* Enable debug logging in the schema-blocks package by changing [this line in Yoast SEO](https://github.com/Yoast/wordpress-seo/blob/b0c471f437123d6415bb45cd9b58eafcb01ecc85/js/src/schema-blocks.js#L4) to:
  ```js
  initialize( LogLevel.DEBUG );
  ```
* Check the browser console.
	* You **should** see debug statements for the schema blocks functionality.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * N/A: behind a feature flag.

## UI changes

## Quality assurance


Fixes #
